### PR TITLE
ENH: Add DiagnosticPreprocessor.

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

**Comment by danielballan** [comment]
This is intended to surface the monitoring, flying, and baseline features in a way that is more accessibly (you don't have to understand preprocessors...) but not tied to SPEC-like plans. The docstring is a bit verbose, in that it basically explains some standard features of Python lists, but some conversations with BL staff have illustrated that this is useful to include.

**Comment by danielballan** [comment]
Address one aspect of https://github.com/NSLS-II/bluesky/issues/677

File: bluesky/plans.py [hunk 1]
```diff
@@ -2998,6 +2998,123 @@ def polling_plan():
...
+    >>> D.baseline.append(some_detector)
+    """
+    def __init__(self, *, baseline=None, monitors=None, flyers=None):
+        if baseline is None:
+            baseline = []
+        if monitors is None:
+            monitors = []
+        if flyers is None:
+            flyers = []
+        self.baseline = baseline
```
Comment (tacaswell) [inline bluesky/plans.py]: Explicitly list-ify here so input of generators and tuples does the right thing.

**Comment by tacaswell** [comment]
We may want to give a bit more thought to tooling around the preprocessor list, but that is out of scope for this PR.

File: bluesky/plans.py (hunk 1 above)
Comment (danielballan) [inline bluesky/plans.py]: Done and force-pushed.

**Comment by stuartcampbell** [comment]
Yeah, I agree about the tooling and some user facing helpers...

**Pull request merged**